### PR TITLE
fix studyJob status suggestionCount mismatch error

### PR DESCRIPTION
**Pull request by hougangliu**
Fixes: #289

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/290)

### CODE REVIEW

File: pkg/controller/studyjob/katib_api_util.go [hunk 1]
```diff
@@ -73,7 +73,6 @@ func initializeStudy(instance *katibv1alpha1.StudyJob, ns string) error {
 		return err
 	}
 	instance.Status.SuggestionParameterID = sPID
-	instance.Status.SuggestionCount += 1
```
Comment (johnugeorge) [inline pkg/controller/studyjob/katib_api_util.go]: why is this removed?

Comment (hougangliu) [reply pkg/controller/studyjob/katib_api_util.go]: SuggestionCount should record how many times the studyJob had called suggestion, and this value should **+= 1** every time after calling suggestion.
Here (initialize studyJob instance and no suggestion called before) instance.Status.SuggestionCount should be 0 (0 by default).

Comment (richardsliu) [reply pkg/controller/studyjob/katib_api_util.go]: @YujiOshima Is this the intended behavior? Should we update the API documentation instead?

Comment (hougangliu) [reply pkg/controller/studyjob/katib_api_util.go]: For now, SuggestionParam["SuggestionCount"] in db change path is [0, 2, 3, ..., requestcount+1].
after this patch its path is [0,1,2,...,requestcount]

Comment (YujiOshima) [reply pkg/controller/studyjob/katib_api_util.go]: @hougangliu Thanks.
> SuggestionParam["SuggestionCount"] in db change path is [0, 2, 3, ..., requestcount+1].
This is not intended behavior, we may need to fix. I'm reviewing and testing this PR.

Comment (johnugeorge) [reply pkg/controller/studyjob/katib_api_util.go]: LGTM

**Comment by YujiOshima** [comment]
@hougangliu Thank you! /approve
